arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2976,7 +2992,18 @@ Result RestReplicationHandler::createBlockingTransaction(aql::QueryId id,
   TRI_ASSERT(isLockHeld(id).is(TRI_ERROR_HTTP_NOT_FOUND));
 
   try {
-    queryRegistry->insert(id, query.get(), ttl, true, true);
+    std::function<void(void)> f =
+      [=]() {
+        try {
+          // Code does not matter, read only access, so we can roll back.
+          queryRegistry->destroy(_vocbase.name(), id, TRI_ERROR_QUERY_KILLED, false);
```

Comment:
This is broken. Capturing by value `[=]` does copy `this`, not `*this`, so just the pointer. Better to copy what you need explicitly. Capturing `*this` is only possible since C++17, and needlessly expensive, so use e.g. `[id, dbName = _vocbase.name()]`.

Reply:
That is because the variables used e.g. `_vocbase` is not in the enclosing scope but `this` is. Therefor an access to `_vocbase` will result in accessing `this->_vocbase`.